www/install/php/Update-21.04.0-beta.1.php

```diff
@@ -18,3 +18,88 @@
  * For more information : contact@centreon.com
  *
  */
+
+include_once __DIR__ . "/../../class/centreonLog.class.php";
+$centreonLog = new CentreonLog();
+
+//error specific content
+$versionOfTheUpgrade = 'UPGRADE - 21.04.0-beta.1: ';
+
+$pearDB = new CentreonDB('centreon', 3, false);
+
+$criteriasConcordanceArray = [
+    'states' => [
+        'acknowledged' => 'Acknowledged',
+        'in_downtime' => 'In downtime',
+        'unhandled_problems' => 'Unhandled',
+    ],
+    'resource_types' => [
+        'host' => 'Host',
+        'service' => 'Service',
+    ],
+    'statuses' => [
+        'OK' => 'Ok',
+        'UP' => 'Up',
+        'WARNING' => 'Warning',
+        'DOWN' => 'Down',
+        'CRITICAL' => 'Critical',
+        'UNREACHABLE' => 'Unreachable',
+        'UNKNOWN' => 'Unknown',
+        'PENDING' => 'Pending'
+    ],
+];
+
+try {
+    $pearDB->beginTransaction();
+    /**
+     * Retrieve user filters
+     */
+    $statement = $pearDB->query(
+        "SELECT `id`, `criterias` FROM `user_filter`"
+    );
+
+    $translatedFilters = [];
+
+    while ($filter = $statement->fetch()) {
+        $id = $filter['id'];
+        $decodedCriterias = json_decode($filter['criterias'], true);
+        // Adding the default sorting in the criterias
+        $decodedCriterias[] = [
+            'name' => 'sort',
+            'type' => 'array',
+            'value' => [
+                'status_severity_code' => "asc"
+            ],
+        ];
+        foreach ($decodedCriterias as $criteriaKey => $criteria) {
+            $name = $criteria['name'];
+            // Checking if the filter contains criterias we want to migrate
+            if (array_key_exists($name, $criteriasConcordanceArray) === true) {
+                foreach ($criteria['value'] as $index => $value) {
+                    $decodedCriterias[$criteriaKey]['value'][$index]['name'] = $criteriasConcordanceArray[$name][$value['id']];
+                }
+            }
+            $translatedFilters[$id] = $decodedCriterias;
```

Comment:
should not be outside foreach ?

Reply:
Nope as I am building a associative array to keep the filter_id to to the next SQL UPDATE request